geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/FilterHelper.scala

```diff
@@ -205,6 +205,41 @@ object FilterHelper {
       }
     }
 
+  /**
+   * Finds the filter with the lowest known cost and returns the rest in the same order they were in.  If
+   * there are multiple filters with the same lowest cost then the first will be selected.  If no filters
+   * have a known cost or if ``s`` is empty then (None, s) will be returned.
+   */
+  def findBest(cost: Filter => Option[Long])(s: Seq[Filter]): (Option[Filter], Seq[Filter]) = {
+    val (best, rest, _) = findBestWithCost(cost, s)
+    (best, rest)
+  }
+
+  private def findBestWithCost(cost: Filter => Option[Long], s: Seq[Filter]): (Option[Filter], Seq[Filter], Option[Long]) = {
```

Comment:
Instead of returning tuples, create a new case class.  Maybe ```CostAnalysis```.  It'll be more readable that way.